src/app/dim-api/reducer.ts

```diff
@@ -382,6 +382,26 @@ export const dimApi = (
   }
 };
 
+/**
+ * DIM accidentally stored some integer settings as strings,
+ * so fix them up here.
+ */
+function fixBadSettingsTypes(settings: Settings) {
+  if (typeof settings.charCol === 'string') {
+    settings = { ...settings, charCol: parseInt(settings.charCol, 10) };
+  }
+  if (typeof settings.charColMobile === 'string') {
+    settings = { ...settings, charColMobile: parseInt(settings.charColMobile, 10) };
+  }
+  if (typeof settings.inventoryClearSpaces === 'string') {
+    settings = { ...settings, inventoryClearSpaces: parseInt(settings.inventoryClearSpaces, 10) };
+  }
+  if (typeof settings.itemSize === 'string') {
+    settings = { ...settings, itemSize: parseInt(settings.itemSize, 10) };
+  }
+  return settings;
+}
```

Comment:
One one hand this could be a loop, on the other it'd mean blindly trusting that these properties are actually numbers in `Settings`, which I didn't love either.